File: articles/warren-mcculloch-walter-pitts-and-neural-logic/haskell/ThresholdUnit.hs
module Main where

thresholdUnit :: [Int] -> [Int] -> Int -> Int
thresholdUnit inputs weights threshold =
  if sum (zipWith (*) inputs weights) >= threshold then 1 else 0

main :: IO ()
main = print (thresholdUnit [1, 1] [1, 1] 2)

File: articles/warren-mcculloch-walter-pitts-and-neural-logic/typescript/threshold_unit.ts
function thresholdUnit(inputs: number[], weights: number[], threshold: number): number {
  const total = inputs.reduce((sum, x, i) => sum + x * weights[i], 0);
  return total >= threshold ? 1 : 0;
}

console.log(thresholdUnit([1, 1], [1, 1], 2));
